src/main/java/kr/allcll/seatfinder/AuthInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package kr.allcll.seatfinder;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+@Component
+public class AuthInterceptor implements HandlerInterceptor {
+
+    private static final String TOKEN_KEY = "token";
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        if (isNotExistToken(request)) {
+            ThreadLocalHolder.SHARED_TOKEN.set(TokenProvider.createToken());
+            return true;
+        }
+        ThreadLocalHolder.SHARED_TOKEN.set(findTokenAtCooke(request));
+        return true;
+    }
+
+    @Override
+    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
+        ModelAndView modelAndView) {
+        if (isNotExistToken(request)) {
+            response.addCookie(new Cookie(TOKEN_KEY, ThreadLocalHolder.SHARED_TOKEN.get()));
+        }
+    }
+
+    private boolean isNotExistToken(HttpServletRequest request) {
+        if (request.getCookies() == null) {
+            return true;
+        }
+        return Arrays.stream(request.getCookies())
+            .noneMatch(cookie -> cookie.getName().equals(TOKEN_KEY));
```

Comment:
35-36번 라인에서 널체크를 해주어서 NPE가 발생하진 않겠지만, 반드시 다음과 같이 작성해야 한다 생각해요!! 동료 개발자가 이 사실을 잘 모르고 35-36을 삭제할 수 있잖아요 😀 수정한다면 44번 라인도 함꼐 수정해야 겠네요! 

```suggestion
        return Arrays.stream(request.getCookies())
            .noneMatch(cookie -> TOKEN_KEY.equals(cookie.getName()));
```